app.js

```diff
@@ -0,0 +1,207 @@
+        </ul>
+        <button type="button" id="${projects[i].idBtn}" onclick="openPopup(${i})" class="see-button-one1">See project</button>
+      </div>`;
+  const nameid = projects[i].id;
+
+  div.querySelector(`#${nameid}`).style.backgroundImage = `url("${projects[i].url}")`;
+  document.querySelector('.middle-section').appendChild(div);
+}
+
+function popcard(i) {
+  const cont = document.querySelector('.popup2');
+  cont.innerHTML = `<h1 id="mob-pop"  >${projects[i].name}</h1>
+  <h2 id="desk-pop" >${projects[i].name1}</h2>
+ <div class="close-button-container2">
+   <button type="button"  class="close-button2" onclick="closePopup()"> &times;</button>
+ </div>
+ 
+ <ul class="cardbtn">
+   <li><button type="button" class="middle-pop">html</button></li>
+   <li> <button type="button" class="middle-pop">bootstrap</button></li>
+   <li><button type="button" class="middle-pop">Ruby On Rails</button></li>
+   
+ </ul>
+ <div class="images-paragraph">
+  <img class="mobile-pop"  src="./assets/images/popup2.png" alt="">
+  <img class="desktop-pop" src="./assets/images/bigpop.png" alt="">
+ <div class="para-btn">
+   <div id="para">
+    <p >${projects[i].description}</p>
+   </div>
+  <div class="popup-btn-container">
+      <button type="button" class="popup-btn1">See Live</button>
+      <button type="button" id="popBtn" class="popup-btn2">See Source</button>
+    </div>
+ </div>
+</div>`;
+}
+
+function openPopup(i) {
+  popcard(i);
+  const pop = document.querySelector('.popup2');
+  pop.style.display = 'flex';
+  document.body.style.overflow = 'hidden';
+  const blur = document.getElementById('middlesec');
+  blur.classList.toggle('active');
+}
+
+function closePopup() {
+  const pop = document.querySelector('.popup2');
+  pop.style.display = 'none';
+  document.body.style.overflow = 'visible';
+}
+
+renderFirstCard();
+
+for (let i = 0; i < 7; i += 1) {
+  renderCards(i);
+}
+
+if (1 === 10) {
+  openPopup();
+  closePopup();
+}
+
+const emailplace = document.getElementById('email-input');
+const fullNameplace = document.getElementById('name-text');
+const commentplace = document.getElementById('user-text');
+function storageSave() {
+  const setData = {
+    name: fullNameplace.value,
+    Email: emailplace.value,
+    Comment: commentplace.value,
+  };
+  localStorage.setItem('StorageData', JSON.stringify(setData));
+}
+
+function getSave() {
+  const getData = JSON.parse(localStorage.getItem('StorageData'));
+  emailplace.value = getData.Email;
+  fullNameplace.value = getData.name;
+  commentplace.value = getData.Comment;
+}
+
+if (!localStorage.getItem('SavedData')) {
+  const SavedData = { name: '', Email: '', Comment: '' };
+  localStorage.setItem('SavedData', JSON.stringify(SavedData));
+} else {
+  getSave();
+}
+const formsc = document.getElementsByClassName('.forms');
+formsc.addEventListener('change', storageSave);
```

Comment:
- [ ] I can see you are trying to implement the saved data here, but it doesn't seem to work as it should. Kindly make sure when the page reload, the form gets the saved data.